### ISSUE TITLE
Input and ListItem.Input do not accept ref prop when using useRef

### ISSUE THREAD

**Issue by ericpowell03** [react-native-elements/react-native-elements#3264]
# What I Did

I have a `<ListItem.Input>` that I would like to `focus()` when another element is clicked. I used `useRef()` to create the ref.

```
function SampleItem(props) {
  const reff = useRef(null);

  return (
    <ListItem>
      <ListItem.Content>
        <Button 
          title="Focus the input"
          onPress={() => {
            reff.current.focus();
          }}
        />
      </ListItem.Content>
      <ListItem.Input 
        ref={reff}
      />
    </ListItem>
  )
}
```

# Expected Behavior

Clicking the button should focus the `ListItem.Input`

# Bug

The current value of the ref remains undefined.

# To Reproduce

Follow code snipped above. If the ListItem.Input is replaced with a plain react native TextInput, the input will properly focus.

# My Environment

| software              | version |
| --------------------- | ------- |
| react-native-elements |  3.4.2       |
| react-native          |    Expo 41.0.0     |

**Comment by entrymaster** [comment]
I would like to work on the bug. Please assign me this issue.

**Comment by arpitBhalla** [comment]
How are you going to do it?

**Comment by entrymaster** [comment]
I think there is a problem with props with Forward Refs in ListItem.Input.tsx . I will try to debug it.

**Comment by entrymaster** [comment]
Can you help me with setting up dev env ?

**Comment by entrymaster** [comment]
Can you guide me in setting up the project? I followed the Contributing Guidelines but didn't work.

**Comment by entrymaster** [comment]
While I was starting example app, I was getting this error:

"While resolving module `@react-native-elements/base/dist/Avatar/Avatar`, the Haste package `@react-native-elements/base` was found. However the module `dist/Avatar/Avatar` could not be found within the package. Indeed, none of these files exist:"

I noticed the actual directory is "src" in example not "dist", I have changed the name from "src" to "dist". Am I allowed to do that ?

**Comment by arpitBhalla** [comment]
No, do `npx lerna run build` in root directory

**Comment by entrymaster** [comment]
Hello Sir, I have got confused with the project structure. In example app every component is being imported from "themed". Should I make changes in packages/base OR packages/themed ? Also how do I test the changed component?

**Comment by arpitBhalla** [comment]
You need to make changes in Base, and test it in example app. So our workflow looks like this

Make change in base -> themed imports base -> themed will use withTheme Hoc -> example app reloads.

I am still working on restructuring the project, so there are few issues, which will be resolved soon.

**Comment by entrymaster** [comment]
The component ListItemInput despite being enclosed in React.forwardRef is generating this warning.

"Warning: Function components cannot be given refs. Attempts to access this ref will fail. Did you mean to use React.forwardRef()?"

It is not detecting forwardRef. Any suggestions ? I'm thinking of changing whole component to class based and passing ref as props to test.

**Comment by arpitBhalla** [comment]
Try importing Component from `@react-native-elements/base`. WIP for Ref in themed pkg

**Comment by entrymaster** [comment]
Yes Its working fine now. Seems ref in themed package is issue. What should I do ?

**Comment by arpitBhalla** [comment]
> WIP for Ref in themed pkg
Our team is working on it, just fix the Input issue ;)

**Comment by entrymaster** [comment]
The issue is fixed. ListItem.Input now is focusable using refs.

**Comment by arpitBhalla** [comment]
Great work!, Create a pull request.

**Comment by entrymaster** [comment]
Update: I have converted the ListItem.Input to a class based component and passed the ref along. Now it is working with themed also.
Please help me create a PR. Should I remove the testing code?

**Comment by arpitBhalla** [comment]
Keep the component as Functional, we are WORKING on themed components, don't remove tests. Follow this guide to create a pull request
https://docs.github.com/en/desktop/contributing-and-collaborating-using-github-desktop/working-with-your-remote-repository-on-github-or-github-enterprise/creating-an-issue-or-pull-request

**Comment by manueldeprada** [comment]
I just wanted to add that if you use React Navigation and wrap a ListItem like 

```
<Link href={{ pathname: '/detailsScreen', 
                                  params: { optId: item.id } }} asChild>
                    <ListItem>
...
```

You get the same error "Function components cannot be given refs". I am importing ListItem from '@rneui/themed' v4.0.0rc8.

**Comment by manueldeprada** [comment]
This is also an issue with Buttons. Expo-router Link component, when used with `asChild` passes a ref. My solution for now to ignote the ref:

```
export const MainButton = forwardRef(({ title, icon, onPress, onClick }, ref) => {
    return <Button
        // ref={ref} //TODO: this should work in a future version of rnneui
        color='none'
        className="m-2.5 bg-black dark:bg-white rounded"
        onPress={onPress}
        onClick={onClick}
    > ...
    </Button>
});
```

That component can be used like:
```
<Link href={{
                    pathname: './z',
                    params: { ... }
                }} asChild>
                    <MainButton title={'x'} icon="y" />
                </Link>
```
